app/change_set_persisters/change_set_persister/create_tombstone.rb

```diff
@@ -7,8 +7,19 @@ def initialize(change_set_persister: nil, change_set:, post_save_resource: nil)
       @change_set_persister = change_set_persister
     end
 
+    def file_set_with_parent?
+      if resource.is_a?(FileSet)
+        resource.try(:original_file) && parent
+      else
+        file_sets = resource.decorate.try(:file_sets)
+        return if file_sets.blank?
+
+        file_sets.first.try(:original_file)
+      end
+    end
+
     def run
-      return unless resource.is_a?(FileSet) && resource.try(:original_file) && parent
+      return unless file_set_with_parent?
```

Comment:
Why do we check for if there are file sets if it's a parent? Wouldn't it just create a tombstone no matter what now?